single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/SvgParameters.java

```diff
@@ -0,0 +1,312 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.sld.svg;
+
+import com.powsybl.diagram.util.ValueFormatter;
+
+import java.util.Locale;
+
+/**
+ *
+ * @author Sophie Frasnedo <sophie.frasnedo at rte-france.com>
+ */
+public class SvgParameters {
+
+    private String prefixId = "";
+    private String undefinedValueSymbol = "\u2014"; //em dash unicode for undefined value
+    private String languageTag = "en";
+    private int voltageValuePrecision = 1;
+    private int powerValuePrecision = 0;
+    private int angleValuePrecision = 1;
+    private int currentValuePrecision = 0;
+    private double busInfoMargin = 0.0; //Can be used as horizontal shifting value for busInfo indicator. Can be negative value
+    private double feederInfosIntraMargin = 10;
+    private double feederInfosOuterMargin = 20;
+    private boolean feederInfoSymmetry = false;
+    private boolean addNodesInfos = false;
+    private boolean useName = false;
+    private double angleLabelShift = 15.;
+    private boolean labelCentered = false;
+    private boolean labelDiagonal = false;
+    private boolean tooltipEnabled = false;
+    private boolean svgWidthAndHeightAdded = false;
+    private SvgParameters.CssLocation cssLocation = SvgParameters.CssLocation.INSERTED_IN_SVG;
+    private boolean avoidSVGComponentsDuplication = false;
+    private String diagramName = null;
+    private boolean drawStraightWires = false;
+    private boolean showGrid = false;
+    private boolean showInternalNodes = false;
+    private boolean displayCurrentFeederInfo = false;
+
+    public SvgParameters() {
+    }
+
+    public enum CssLocation {
+        INSERTED_IN_SVG, EXTERNAL_IMPORTED, EXTERNAL_NO_IMPORT
+    }
+
+    public SvgParameters(SvgParameters other) {
+        this.prefixId = other.prefixId;
```

Comment:
A general comment: it is really important in an API to check for null parameters.